x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -159,67 +164,70 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 				continue
 			}
 
-			if namespaceResourceType.metricNames != nil && namespaceResourceType.dimensions == nil {
-				for _, listMetric := range listMetricsOutput {
-					if exists, _ := aws.StringInSlice(*listMetric.MetricName, namespaceResourceType.metricNames); !exists {
-						continue
-					}
-					filteredMetricWithStatsTotal = append(filteredMetricWithStatsTotal,
-						metricsWithStatistics{
-							cloudwatchMetric: listMetric,
-							statistic:        namespaceResourceType.statistic,
-						})
-
-					exists, _ := aws.StringInSlice(namespaceResourceType.resourceTypeFilter, resourceTypes)
-					if namespaceResourceType.resourceTypeFilter != "" && !exists {
-						resourceTypes = append(resourceTypes, namespaceResourceType.resourceTypeFilter)
+			for _, metricD := range metricDetails {
```

Comment:
minor, but maybe `metricDetail` could be more readable here? :)

Reply:
Yeah this naming is bad. The _Details_ I'm referring here is actually configuration details per each namespace in cloudwatch metricset config. So I will rename _metricDetails_ to _namespaceDetails_ and _metricD_ to _configPerNamespace_.